base/array.jl

```diff
@@ -1723,48 +1723,61 @@ findlast(testf::Function, A) = findprev(testf, A, endof(A))
 """
     find(f::Function, A)
 
-Return a vector `I` of the linear indexes of `A` where `f(A[I])` returns `true`.
+Return a vector `I` of the indices or keys of `A` where `f(A[I])` returns `true`.
 If there are no such elements of `A`, return an empty array.
 
+Indices or keys are of the same type as those returned by [`keys(A)`](@ref)
+and [`pairs(A)`](@ref) for `AbstractArray` and `Associative` objects,
+and are linear indices of type `Int` for other iterables.
+
 # Examples
 ```jldoctest
-julia> A = [1 2 0; 3 4 0]
-2×3 Array{Int64,2}:
- 1  2  0
- 3  4  0
+julia> x = [1, 3, 4]
+3-element Array{Int64,1}:
+ 1
+ 3
+ 4
 
-julia> find(isodd, A)
+julia> find(isodd, x)
 2-element Array{Int64,1}:
  1
  2
 
+ julia> A = [1 2 0; 3 4 0]
+ 2×3 Array{Int64,2}:
+  1  2  0
+  3  4  0
+
+ julia> find(isodd, A)
+ 2-element Array{CartesianIndex{2},1}:
+  CartesianIndex(1, 1)
```

Comment:
A nice feature is that with https://github.com/JuliaLang/julia/pull/24651/ the `CartesianIndex` part won't be repeated for each element, making the returned object quite natural.